File: src/main/java/com/example/demo/mapper/TimesheetEntryMapper.java
package com.example.demo.mapper;

import com.example.demo.model.TimesheetEntry;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;

import java.time.LocalDate;
import java.util.List;

@Mapper
public interface TimesheetEntryMapper {
    TimesheetEntry selectByUserAndDate(
            @Param("userName") String userName,
            @Param("workDate") LocalDate workDate
    );

    List<TimesheetEntry> selectByUserAndRange(
            @Param("userName") String userName,
            @Param("from") LocalDate from,
            @Param("to") LocalDate to
    );

    int insert(TimesheetEntry entry);

    int updateTimes(TimesheetEntry entry);

    int updateTimesForce(TimesheetEntry entry);

    int updateNote(@Param("id") Long id, @Param("note") String note);
}

File: src/main/kotlin/com/example/demo/mapper/TimesheetEntryMapper.kt
package com.example.demo.mapper

import com.example.demo.model.TimesheetEntry
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import java.time.LocalDate

@Mapper
interface TimesheetEntryMapper {
    fun selectByUserAndDate(
        @Param("userName") userName: String,
        @Param("workDate") workDate: LocalDate
    ): TimesheetEntry?

    fun selectByUserAndRange(
        @Param("userName") userName: String,
        @Param("from") from: LocalDate,
        @Param("to") to: LocalDate
    ): List<TimesheetEntry>

    fun insert(entry: TimesheetEntry): Int
    fun updateTimes(entry: TimesheetEntry): Int // break/duration/working 追加後も同名で利用
    fun updateTimesForce(entry: TimesheetEntry): Int
    fun updateNote(@Param("id") id: Long, @Param("note") note: String): Int
}
